pkg/metrics/collector/fs.go

```diff
@@ -41,6 +102,36 @@ func (f *FsMetricsCollector) Collect() {
 	}
 }
 
+func (i *InflightMetricsVecCollector) Collect() {
+	if i.MetricsVec == nil {
+		log.L.Warnf("can not collect inflight metrics: Metrics is nil")
+		return
+	}
+
+	hungIOMap := make(map[string]uint64)
+	recordedHungIOMap := make(map[uint64]uint64)
+	nowTime := time.Now()
+	for _, daemonInflightIOMetrics := range i.MetricsVec {
+		for _, inflightIOMetric := range daemonInflightIOMetrics.Values {
+			if value, ok := i.RecordedMetrics[inflightIOMetric.Unique]; ok {
+				recordedHungIOMap[inflightIOMetric.Unique] += value
+			} else {
+				elapsed := nowTime.Sub(time.Unix(int64(inflightIOMetric.TimestampSecs), 0))
+				if elapsed >= i.HungIOPeriod {
+					hungIOMap[OPCodeMap[inflightIOMetric.Opcode]]++
+					recordedHungIOMap[inflightIOMetric.Unique]++
+					log.L.Debugf("Record hung IO, Inode: %v, Opcode: %v, Unique: %v, Elapsed: %v",
+						inflightIOMetric.Inode, OPCodeMap[inflightIOMetric.Opcode], inflightIOMetric.Unique, elapsed)
+				}
+			}
+		}
+	}
+	i.RecordedMetrics = recordedHungIOMap
+	for opcode, value := range hungIOMap {
+		data.TotalHungIO.WithLabelValues(opcode).Add(float64(value))
```

Comment:
The metric should be Gauge. Hung io can disappear when the backend remote request is responded.

Reply:
Got it. I have replaced it with `Gauge`.